### ISSUE TITLE
do we omit density calc for all points that don't have 2 trees in two different quadrants?

### ISSUE THREAD

**Issue by paciorek** [PalEON-Project/PLS_products#27]
Seemingly @SimonGoring 's code omits cases where we don't have two quadrants (and it's stated in the paper), but this appears in the code:

```
  #  There are 16,197 points with an NA for the azimuth, but with two recorded
  #  distances, these are pretty much all in Michigan.  These are the ones
  #  we need to change:
  
  two.quads[((two.quads < 2 & (is.na(azim[,1]) | is.na(azim[,2]))) &
    !(is.na(dist[,1]) | is.na(dist[,2])))] <- 2
```

let's clarify why we include trees with a missing azimuth if they have two valid distances.

**Comment by SimonGoring** [comment]
Yes, this differentiates between any plot where both trees are sampled in the same quadrant and trees that seem to have two 

```

**Comment by paciorek** [comment]
Per email with Charlie, we will include density calcs for all cases except when we have clear info that the two nearest trees are in the same quadrant. I.e. cases with one or two missing 360 degree azimuths will be included.

**Comment by paciorek** [comment]
@paciorek still wants to check the number of cases where one or more azimuths are missing. Leaving this open as a reminder.

**Comment by paciorek** [comment]
Per check in with Charlie, we should exclude same-quadrant points for both 2nQ and Pair points. As of April 2019, there are 935 such 2nQ and 8670 such Pair.

All the mathematics for density we use assumes that each tree represents the nearest in its own sector. This is nominally quadrants for 2nQ corners and if 2 trees are in the same quadrant, it is by definition not 2nQ.  For Pairs, the assumption is relaxed, not requiring quadrant sectors, but the geometry is "corrected" by theta which is determined after removing corners with "duplicate" quadrants. Thus with our present analyses, we cannot incorporate these "duplicate" quadrants. Technically each of these corners can be treated as a k-1 tree corner (representing the number of quadrants filled) by removing the farther tree in the quadrant.  This then creates more 1-tree or 3-tree corners which are problematic in their own right. Way back Simon faced this question and simply removed any corner with duplicate quadrants.  We should probably follow this rule, despite loss of data, since the runaround is complex.
